src/cms/models/users/user.py

```diff
@@ -0,0 +1,240 @@
+"""
+Custom user model that is used instead of the default Djang user model
+"""
+
+import logging
+
+from django.db import models
+from django.utils import timezone
+from django.utils.translation import ugettext_lazy as _
+from django.contrib.auth.models import AbstractUser
+from django.contrib.auth.base_user import BaseUserManager
+
+from .organization import Organization
+from ..chat.chat_message import ChatMessage
+from ..regions.region import Region
+
+logger = logging.getLogger(__name__)
+
+
+class UserManager(BaseUserManager):
+    """
+    Replacement for the default UserManager
+    """
+
+    def create_user(
+        self, email=None, username=None, password=None, staffuser=False, superuser=False
+    ):
+        """
+        Creates and saves a User with the given email and password.
+
+        :param email: user's e-mail address
+        :type email: str
+        :param username: user's name
+        :type username: str
+        :param password: user's password
+        :type password: str
+        :param staffuser: staff status
+        :type staffuser: bool
+        :param superuser: superuser status
+        :type superuser: bool
+
+        :raises ValueError: raised if required email parameter is not passed
+
+        :return: created user instance
+        :rtype: ~cms.models.users.user.User
+        """
+        if not email:
+            raise ValueError("Users must have an email address")
+
+        user = self.model(
+            email=self.normalize_email(email),
+        )
+
+        user.username = username
+        user.set_password(password)
+
+        user.is_staff = staffuser
+        user.is_superuser = superuser
+
+        user.save(using=self._db)
+        return user
+
+    def create_staffuser(self, email=None, username=None, password=None):
+        """
+        Creates and saves a staff user with the given email and password.
+
+        :param email: user's e-mail address
+        :type email: str
+        :param username: user's name
+        :type username: str
+        :param password: user's password
+        :type password: str
+
+        :return: created user instance
+        :rtype: ~cms.models.users.user.User
+        """
+        return self.create_user(
+            email=email, username=username, password=password, staffuser=True
+        )
+
+    def create_superuser(self, email=None, username=None, password=None):
+        """
+        Creates and saves a superuser with the given email and password.
+
+        :param email: user's e-mail address
+        :type email: str
+        :param username: user's name
+        :type username: str
+        :param password: user's password
+        :type password: str
+
+        :return: created user instance
+        :rtype: ~cms.models.users.user.User
+        """
+        return self.create_user(
+            email=email,
+            username=username,
+            password=password,
+            superuser=True,
+            staffuser=True,
+        )
```

Comment:
Actually, I don't think we need a custom user manager. Since we use all default fields of the default [AbstractUser](https://github.com/django/django/blob/4ffada36097ceccfd6f0d358217ac3c40d2a729c/django/contrib/auth/models.py#L321), we can as well go with the default [UserManager](https://github.com/django/django/blob/4ffada36097ceccfd6f0d358217ac3c40d2a729c/django/contrib/auth/models.py#L129)... As I understand [the documentation](https://docs.djangoproject.com/en/3.2/topics/auth/customizing/#writing-a-manager-for-a-custom-user-model), we only need a custom manager, if we use a completely different user model without the default fields `username` and  `email` (in other words, if our custom user model would inherit from `AbstractBaseUser` instead of `AbstractUser`). Since the default `UserManager` already has support for `extra_fields`, adding custom fields is not a problem.
The only difference might be that the default `create_user()` does not require an email address, but this function isn't used anywhere except the tests. And e.g. via the admin interface, it's possible to create a user without email anyways (even with the custom user manager), so I don't see an advantage of the "more safe" `create_user()`-method...
(and also the `create_staffuser()`-method is only present on the custom manager, but I don't see an important use case for this either...)

Reply:
I'll right. I tried to keep the test as is and lost focus. Seems like the better solution to change the test, yes ;-)